Asterix/Optical_System_functions.py

```diff
@@ -1387,27 +1434,29 @@ def creatingpushact_inpup(self,
         ----------
         load_fits : bool, default = False if true, we do not measure creatingpushact_inpup fits, we load it
         save_fits : bool, default = False if true, we save the creatingpushact_inpup fits for future use
-        Model_local_dir: directory to save things you can measure yourself 
+        Model_local_dir: directory to save things you can measure yourself
                     and can save to save time
         Returns
         ------
         pushact_inpup : Map of the complex phase induced in pupil plane
         -------------------------------------------------- """
 
-        Name_pushact_fits = self.Name_DM + "_PushActInPup_radpup" + str(
+        Name_pushact_inpup_fits = self.Name_DM + "_PushActInPup_radpup" + str(
             int(self.pradDM)) + "_dimpuparray" + str(
-                int(self.dim_overpad_pupil))
-
-        if (load_fits
-                == True) or (self.creating_pushact == False
-                             and os.path.exists(Model_local_dir +
-                                                Name_pushact_fits + '.fits')):
+                int(self.dim_overpad_pupil)) + "_z" + str(
+                    int(self.z_position * 1000)) + 'mm'
+
+        if (load_fits is True) and (
+                os.path.exists(Model_local_dir + Name_pushact_inpup_fits +
+                               '_inPup_real.fits')
+        ) and (os.path.exists(Model_local_dir + Name_pushact_inpup_fits +
+                              '_inPup_imag.fits')):
             DM_pushact_inpup_real = fits.getdata(
                 os.path.join(Model_local_dir,
-                             Name_pushact_fits + '_inPup_real.fits'))
+                             Name_pushact_inpup_fits + '_inPup_real.fits'))
             DM_pushact_inpup_imag = fits.getdata(
                 os.path.join(Model_local_dir,
-                             Name_pushact_fits + '_inPup_imag.fits'))
+                             Name_pushact_inpup_fits + '_inPup_imag.fits'))
```

Comment:
Comment about the question that is below "Do we need a pupil?"
I believe we do for the Fresnel propagation. Need to think about it.

Reply:
yes, this is one of the question we need to think about, I have been thinking about it. this is #1 on my list for next wednesday meeting. I think we do, but there are other problems !